engine/access/rpc/backend/transactions_local_data_provider.go

```diff
@@ -21,6 +21,9 @@ import (
 	"github.com/onflow/flow-go/storage"
 )
 
+// ErrTransactionNotInBlock represents an error indicating that the transaction is not found in the block.
+var ErrTransactionNotInBlock = errors.New("transaction not in block")
```

Comment:
Why do we need this sentinel? To me it seems we can return `storage.ErrNotFound` instead of this.

Reply:
I requested this. It's because the methods used to lookup a transaction in a block need to lookup each collection in the guarantee list, then check if the transaction exists in one of the collections. Collections are synced by the Access node after the block is finalized, so it's a common for there to be collections that aren't available yet when doing this lookup for non-sealed blocks.

This makes it easy for the caller to differentiate between a block that has missing collections, and one where all collections were indexed, but the tx just wasn't included.